library/packages/src/modules/PackageAI.rb

```diff
@@ -29,123 +29,101 @@
 # $Id$
 require "yast"
 
+Yast.import "PackagesProposal"
+
 module Yast
   class PackageAIClass < Module
     def main
       textdomain "base"
-
-      @toinstall = []
-      @toremove = []
-
-      @last_op_canceled = false
-
-      Yast.include self, "packages/common.rb"
-
-      # default value of settings modified
-      @modified = false
-    end
-
-    # Function sets internal variable, which indicates, that any
-    # settings were modified, to "true"
-    def SetModified
-      @modified = true
-
-      nil
     end
 
-    # Functions which returns if the settings were modified
-    # @return [Boolean]  settings were modified
-    def GetModified
-      @modified
-    end
-
-    def DoInstall(packages)
-      packages = deep_copy(packages)
-      @toinstall = Convert.convert(
-        Builtins.union(@toinstall, packages),
-        from: "list",
-        to:   "list <string>"
-      )
-      @toremove = Builtins.filter(@toremove) do |p|
-        !Builtins.contains(packages, p)
+    # Install and remove packages in one go
+    #
+    # @note In AutoYaST, packages are added or removed from the
+    #       {Yast::PackagesProposalClass packages proposal} instead of actually
+    #       installing or removing them from the system.
+    #
+    # @param toinstall [Array<String>] Name of the packages to install
+    # @param toremove [Array<String>] Name of the packages to remove
+    # @return [Boolean] true on success
+    #
+    # @see Yast::PackageClass#DoInstallAndRemove
+    def DoInstallAndRemove(toinst, torem)
+      if !toinst.empty?
+        Yast::PackagesProposal.AddResolvables("autoyast", :package, toinst)
+        Yast::PackagesProposal.RemoveTaboos("autoyast", toinst) # FIXME: should be done by PackagesProposal
```

Comment:
Well, tricky question here. What will happen if different unique id taboo package?

Reply:
Then it is tabooed.